### PR TITLE
fix(cli): Return exit code on Argo template lint command

### LINKED ISSUE #4282: `argo template lint` Always Exits With 0 Exit Code When Validating a Directory
## Summary 

When `argo template lint` is used to lint a directory containing workflows, it currently always exits with a 0 exit code even if errors are encountered. Exit codes are one way that shells implement control flow, and always returning a zero exit code is similar to an HTTP endpoint always returning `200 OK`.

## Expected Behavior

When `argo template lint` is used to validate a directory containing workflows or workflow templates, it should exit with a nonzero exit code in the case that validation fails for one or more workflows or workflow templates.

## Diagnostics

```
$ argo version
argo: latest+a00a8f1.dirty
  BuildDate: 2020-10-08T01:44:49Z
  GitCommit: a00a8f141c221f50e397aea8f86a54171441e395
  GitTreeState: dirty
  GitTag: v2.11.3
  GoVersion: go1.15.2
  Compiler: gc
  Platform: darwin/amd64
```

./abc/abc.yaml:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: example-whalesay2
spec:
  entrypoint: jklfdsa
```
```
$ argo template lint ./abc
ERRO[0000] template name 'jklfdsa' undefined            
WorkflowTemplate manifests validated

$ echo $?
0
```

This does work as expected in the case that only a single file is validated:
```
$ argo template lint ./abc/abc.yaml
ERRO[0000] template name 'jklfdsa' undefined            
FATA[0000] Errors encountered in validation             

$ echo $?
1
```

## Workaround
Until this issue is resolved, validate each file individually.
```
for f in $(find './templates' -iname '*.yaml'); do
  argo template lint "$f";
done
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by sarabala1979**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
* [ ] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
Fixes #4282

### CODE REVIEW

File: cmd/argo/commands/template/lint.go [hunk 1]
```diff
@@ -77,7 +77,11 @@ func ServerSideLint(args []string, strict bool) error {
 			}
 			return nil
 		}
-		return filepath.Walk(args[0], walkFunc)
+		err := filepath.Walk(args[0], walkFunc)
+		if err != nil {
```
Comment (alexec) [inline cmd/argo/commands/template/lint.go:81]: test?

**Review by alexec — CHANGES_REQUESTED** [review]
needs tests

File: cmd/argo/commands/template/lint.go (hunk 1 above)
Comment (sarabala1979) [reply cmd/argo/commands/template/lint.go:81]: Currently, without a mock APIServer we can't write the test. The test will be added in CLI test framework PR

Comment (alexec) [reply cmd/argo/commands/template/lint.go:81]: e2e test?

Comment (sarabala1979) [reply cmd/argo/commands/template/lint.go:81]: added e2e test

**Review by alexec — APPROVED** [review]

**Pull request merged**